eval/src/vespa/eval/eval/aggr.h

```diff
@@ -120,6 +121,32 @@ template <typename T> class Max {
     constexpr T result() const { return _max; }
 };
 
+template <typename T> class Median {
+private:
+    std::vector<T> _seen;
+public:
+    constexpr Median() : _seen() {}
+    constexpr Median(T value) : _seen({value}) {}
+    constexpr void sample(T value) { _seen.push_back(value); }
+    constexpr void merge(const Median &rhs) {
+        for (T value: rhs._seen) {
+            _seen.push_back(value);
+        }
+    };
+    constexpr T result() const {
+        if (_seen.empty()) {
+            return std::numeric_limits<T>::quiet_NaN();
+        }
+        std::vector<T> tmp = _seen;
+        std::sort(tmp.begin(), tmp.end());
+        size_t cnt = tmp.size();
+        if ((cnt % 2) == 0) {
+            return ((tmp[cnt/2] + tmp[(cnt-1)/2]) / T{2});
```

Comment:
if you want, you can always do this, because if "cnt" is odd then (cnt/2) == (cnt-1)/2 and (x+x)/2 == x

Reply:
isn't std::nth_element, with an extra std::max_element or std::min_element in case of even number of elements cheaper ?

Reply:
@arnej27959 : yes, I also thought of that, but could not get myself to do it, even if the performance will be 'the same'
@toregge : I went for simple over fast, since it will be slow no matter what. Also, I am still not friends with nth_element (we have history). More importantly, your comment reminds me that sort will not work with NaN values. Need a new sort that works with NaN, I guess... hmm